FFmpegImage: Fix CID 1343330 and CID 1343329

fritsch opened the pull request:
@ace20022 for review please. The second hunk will most likely only warn on architectures where size_t is 64 bit see the coverity report.

fritsch committed:
FFmpegImage: Fix CID 1343330 and CID 1343329

FernetMenta commented on xbmc/guilib/FFmpegImage.cpp:
```diff
@@ -225,7 +225,7 @@ bool CFFmpegImage::Decode(unsigned char * const pixels, unsigned int width, unsi
     return false;
 
-  if (!m_pFrame || !m_pFrame->data)
+  if (!m_pFrame || !m_pFrame->data[0])
```
you can't access data[0] if you are not sure if data != nullptr. data is a pointer to an array of pointers

fritsch replied:
I think I can: https://ffmpeg.org/doxygen/2.7/structAVFrame.html

fritsch replied:
```
 CID 1343330:    (NO_EFFECT)
>>>     Comparing an array to null is not useful: "!this->m_pFrame->data", since the test will always evaluate as true.
228       if (!m_pFrame || !m_pFrame->data)
```

That's not a simple ptr, it's ptr* to something[something_else]

FernetMenta replied:
right, I mixed it up with extended_data

fritsch committed:
FFmpegImage: Care when have jpeg as input
v1: Initial implementation
v2: Use av_frame_get_color_range instead of manual parsing

ace20022 commented:
Okay, int64_t should really be enough.

fritsch commented:
@ace20022 I will close this one and add the commits to the other PR which touches the whole class quite heavily.
